### PR TITLE
Completed TC01 30/8/2021

### PR DESCRIPTION
**Pull request by HoanVanHuynh**

### CODE REVIEW

File: src/test/java/tests/LoginTests.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import common.Constants;
+import helpers.LogHelper;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import page_objects.BasePage;
+import page_objects.LoginPage;
+
+public class LoginTests extends BaseTest {
+
+    private BasePage basePage = new BasePage();
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java]: don't call BasePage directly in test case

File: src/test/java/tests/LoginTests.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        basePage.clickLoginTab();
+        LogHelper.info("Enter valid Email and Password and click on Login button");
+        loginPage.login(Constants.USERNAME, Constants.PASSWORD);
+
+        LogHelper.info("Get welcome user message");
+        String actualMessage = loginPage.getWelcomeMessage();
+        String expectedMessage = "Welcome " + Constants.USERNAME;
+
+        Assert.assertEquals(actualMessage, expectedMessage, "Welcome user message is not displayed as expected");
+
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java]: unneeded downline

File: src/main/java/page_objects/BasePage.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    // Locators
+    private final By tabLogin = By.cssSelector("a[href$='Login.cshtml']");
+
+    // Elements
+    private WebElement getTabLogin() {
+        return DriverHelper.getDriver().findElement(tabLogin);
+    }
+
+    // Methods
+
```
Comment (tanmle) [inline src/main/java/page_objects/BasePage.java]: unneeded downline

File: src/test/java/tests/LoginTests.java (hunk 1 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/LoginTests.java]: thank you, I will fix that

File: src/test/java/tests/LoginTests.java (hunk 2 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/LoginTests.java]: Thank you so much, I'll pay attention to it

File: src/main/java/page_objects/BasePage.java (hunk 3 above)
Comment (HoanVanHuynh) [reply src/main/java/page_objects/BasePage.java]: Thank you so much, I'll pay attention to it

File: src/test/java/tests/LoginTests.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public void tc01_LoginWithValidInformation() {
+        LogHelper.info("Click on Login tab");
+        loginPage.clickLoginTab();
+        LogHelper.info("Enter valid Email and Password and click on Login button");
+        loginPage.login(Constants.USERNAME, Constants.PASSWORD);
+
+        LogHelper.info("Get welcome user message");
+        String actualMessage = loginPage.getWelcomeMessage();
+        String expectedMessage = "Welcome " + Constants.USERNAME;
+        Assert.assertEquals(actualMessage, expectedMessage, "Welcome user message is not displayed as expected");
```
Comment (MinhQuan119) [inline src/test/java/tests/LoginTests.java:26]: Add another log for assert. It does not belong to "Get welcome user message" step

**Review by anhlvt — APPROVED** [review]

File: src/main/java/page_objects/LoginPage.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+public class LoginPage extends BasePage{
+
+    // Locators
+    private final By txtUsername = By.id("username");
+    private final By txtPassword = By.id("password");
+    private final By btnLogin = By.cssSelector("input[value='Login']");
+    private final By lblWelcomeMessage = By.cssSelector("div[class='account'] > strong");
+
+
```
Comment (MinhQuan119) [inline src/main/java/page_objects/LoginPage.java]: The space between Locators and Elements on this page is 2 lines while on BasePage, it is 1 line. Please make them the same format

**Review by MinhQuan119 — CHANGES_REQUESTED** [review]

File: src/test/java/tests/LoginTests.java (hunk 4 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/LoginTests.java:26]: Thank you so much. I will delete that line

File: src/main/java/page_objects/LoginPage.java (hunk 5 above)
Comment (HoanVanHuynh) [reply src/main/java/page_objects/LoginPage.java]: Thank you so much, I'll pay attention to it

File: src/main/java/page_objects/LoginPage.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        ElementHelper.scrollTo(this.getTxtUsername());
+        this.getTxtUsername().sendKeys(user);
+        this.getTxtPassword().sendKeys(password);
+        this.getBtnLogin().click();
+    }
+
+    public String getWelcomeMessage() {
+        return this.getLblWelcomeMessage().getText();
+    }
+
```
Comment (tanmle) [inline src/main/java/page_objects/LoginPage.java]: unneeded donwline

File: src/main/java/page_objects/BasePage.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    // Elements
+    private WebElement getTabLogin() {
+        return DriverHelper.getDriver().findElement(tabLogin);
+    }
+
+    // Methods
+    public void clickLoginTab() {
+        this.getTabLogin().click();
+    }
+
```
Comment (tanmle) [inline src/main/java/page_objects/BasePage.java]: Unneeded downline

Comment (HoanVanHuynh) [reply src/main/java/page_objects/BasePage.java]: I'm so sorry about Unneeded downline error. This error will not happen in the future

File: src/main/java/page_objects/LoginPage.java (hunk 6 above)
Comment (HoanVanHuynh) [reply src/main/java/page_objects/LoginPage.java]: I'm so sorry about Unneeded downline error. This error will not happen in the future

**Review by MinhQuan119 — APPROVED** [review]

**Pull request merged**